src/daft-logical-plan/src/optimization/optimizer.rs

```diff
@@ -93,6 +94,12 @@ impl Optimizer {
     pub fn new(config: OptimizerConfig) -> Self {
         let mut rule_batches = Vec::new();
 
+        // we want to simplify expressions first to make the rest of the rules easier
+        rule_batches.push(RuleBatch::new(
+            vec![Box::new(SimplifyExpressionsRule::new())],
+            RuleExecutionStrategy::FixedPoint(Some(3)),
+        ));
+
```

Comment:
I would put this under the rewrite rules since in my mind, the plan is in an invalid state before the rewrite rules are applied, so the rewrite rules should not rely on any optimizer rules but the optimizer rules can make assumptions about the validity of the plan.